ppp_hal/requesthandler.py

```diff
@@ -40,22 +40,91 @@ def query(query, fields):
         mc.set(key, r, time=Config().memcached_timeout)
     return r
 
+PAPER_FIELDS = ('abstract_s', 'releasedDate_s', 'modifiedDateY_i',
+        'uri_s', 'halId_s', 'title_s', 'authFullName_s', 'arxivId_s',
+        'authFirstName_s', 'authLastName_s', 'authOrganism_s',
+        'labStructName_s', 'version_i', 'language_s')
+
+def graph_from_paper(paper):
+    same_as = list(filter(bool, (
+        paper['halId_s'],
+        paper.get('arxivId_s', None),
+        )))
+    organizations = paper.get('authOrganism_s', []) + \
+            paper.get('labStructName_s', [])
+    authors = [{'@context': 'http://schema.org',
+                '@type': 'Person',
+                'name': fullname,
+                'givenName': firstname,
+                'familyName': lastname,
+               }
+               for (fullname, firstname, lastname) in
+               zip(paper['authFullName_s'], paper['authFirstName_s'],
+                   paper['authLastName_s'])]
+    d = {
+            '@type': 'ScholarlyArticle',
+            '@context': 'http://schema.org',
+            'description': paper.get('abstract_s', None),
+            'datePublished': paper['releasedDate_s'],
+            'dateModified': paper['modifiedDateY_i'],
+            '@id': paper['uri_s'],
+            'isSameAs': paper['halId_s'],
```

Comment:
I don't think that schema:isSameAs exits. You should use URIs as identifiers and not simple strings like "hal-00600993"